wire-ios-data-model/Source/MLS/Migration/ProteusToMLSMigrationCoordinator.swift

```diff
@@ -148,6 +147,42 @@ public class ProteusToMLSMigrationCoordinator: ProteusToMLSMigrationCoordinating
         return .canStart
     }
 
+    /// This method is responsible for migrating all `mixed` group conversations to `mls`.
+    /// It evaluates each conversation to determine if the migration needs to be finalized (i.e: updating the protocol from `mixed` to `mls`)
+    /// or if it should first join the corresponding MLS group.
+    func migrateOrJoinGroupConversations() async {
+        do {
+            let conversations = try ZMConversation.fetchAllTeamGroupConversations(
+                messageProtocol: .mixed,
+                in: context
+            )
+
+            guard let mlsService = context.mlsService else {
+                return logger.warn("can't migrate conversations to mls: missing `mlsService`")
+            }
+
+            for conversation in conversations {
+                do {
+                    guard let groupID = conversation.mlsGroupID else {
+                        return logger.warn("can't migrate conversation to mls: missing `groupID`")
```

Comment:
I would suggest to extract all mlsGroupID from the context and loop from it then

Reply:
Done!